### PR TITLE
filter shortcuts view by a string term

### PR DESCRIPTION
**Pull request by antis81**
implements #225 

@tsujan Done! (Effort was unexpectedly low. Basically a little revisiting of QSortFilterProxyModel documenation! :smile:)

### CODE REVIEW

File: config/main_window.cpp [hunk 1]
```diff
@@ -69,11 +69,18 @@ MainWindow::MainWindow(QWidget *parent)
...
     actions_TV->setModel(mSortFilterProxyModel);
 
     mSelectionModel = new QItemSelectionModel(actions_TV->model());
     actions_TV->setSelectionModel(mSelectionModel);
 
+    connect(filter_LE, &QLineEdit::textChanged, [this]{
+      auto filterTerm = filter_LE->text();
+      mSortFilterProxyModel->setFilterRegularExpression(filterTerm);
+    });
+
```
Comment (tsujan) [inline config/main_window.cpp:83]: Glad to see such a fast response!

Your work seems good to me but please change the above connection to:

```c++
connect(filter_LE, &QLineEdit::textChanged, mSortFilterProxyModel, &QSortFilterProxyModel::setFilterWildcard);
```

Many users aren't familiar with regex and may see strange results when, for example, they type `alt*`. And there's no need to a lambda.

File: config/main_window.ui [hunk 2]
```diff
@@ -17,32 +17,43 @@
...
-       <attribute name="verticalHeaderVisible">
-        <bool>false</bool>
-       </attribute>
-      </widget>
+      <layout class="QVBoxLayout" name="verticalLayout_3">
+       <item>
+        <widget class="QLineEdit" name="filter_LE">
+         <property name="placeholderText">
+          <string>Search</string>
+         </property>
```
Comment (tsujan) [inline config/main_window.ui:25]: Please add this here:

```xml
         <property name="clearButtonEnabled">
          <bool>true</bool>
         </property>
```

**Comment by stefonarch** [comment]
Am I missing something? Searching for shortcuts doesn't work here:

**Comment by tsujan** [comment]
> Am I missing something? Searching for shortcuts doesn't work here:

It will after @antis81 makes the changes that I mentioned in the review.

**Comment by antis81** [comment]
I think we don't need any complex parser filtering here. A simple fixed string search perfectly does the job.

**Review by tsujan — APPROVED** [review]
OK, thanks!

In future, when you make a PR for an issue or feature request, please add "fixes ISSUE_ADDRESS" or "closes ISSUE_ADDRESS" to the PR's comment to link the PR with the report.

**Comment by tsujan** [comment]
@stefonarch 
Would you please test it again?

**Comment by stefonarch** [comment]
GTM, working fine now. Thanks @antis81 !

**Pull request merged**